tor/main.py

```diff
@@ -5,6 +5,7 @@
 from tor_core.helpers import run_until_dead
 from tor_core.initialize import build_bot
 
+import tor
```

Comment:
Let's mix this import with the one below it

Reply:
```python
from tor import __version__, __SELF_NAME__, __BOT_NAME__
```

Reply:
The reason these are separate is because we _want_ to mutate global state of `__SELF_NAME__`. If we refer to it as `__SELF_NAME__` only, it's just like any other variable we're shadowing (redefining over the top of it, not affecting the original) whereas assigning it like this: `tor.__SELF_NAME__ = 'foo'`, it affects the reference of it instead. It's assigning a value to the attribute of a global item that isn't being mutated directly, but rather just one of its attributes' values.

Difficult to explain, but this is some magic where we _want_ to affect global state and we can't do it if we do what you're suggesting.

Reply:
I am mildly horrified that this is a thing, but I understand what you're explaining it. Rock on 👍